pkg/plugins/golang/declarative/v1/scaffolds/internal/templates/types.go

```diff
@@ -33,18 +33,31 @@ type Types struct {
 	machinery.MultiGroupMixin
 	machinery.BoilerplateMixin
 	machinery.ResourceMixin
+
+	IsLegacyLayout bool
 }
 
 // SetTemplateDefaults implements file.Template
 func (f *Types) SetTemplateDefaults() error {
 	if f.Path == "" {
 		if f.MultiGroup {
-			f.Path = filepath.Join("apis", "%[group]", "%[version]", "%[kind]_types.go")
+			if f.IsLegacyLayout {
```

Comment:
Hi @varshaprasad96 I understand your suggestion here.
The if/else here is also because we need create an diff path for multigroup
I go along that it has a hall for improvements but them the refac will be bigger.

Could we agree in keep it as it is today already ( indeed before these changes ) and see if we can improve in a follow-up?

Reply:
Sure, we can do in a follow up. Just wanted to make sure that we make the conditionals as modular as possible, so that its easier to refactor when we remove support to the previous format ever in future.